### PR TITLE
Various code fixes

### PR DESCRIPTION
**Pull request by DenysVuika**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

- add missing await to various tests (based on IntelliJ code analysis)
- fix typings for certain scenarios
- remove unused variables
- fix bug with incorrect i18n in unit tests
- fix bug with incorrect rule evaluator code (comma instead of `&&`)
- add missing variable initializers

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: e2e/suites/extensions/ext-document-list.test.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ describe('Extensions - DocumentList presets', () => {
     const expectedColumns = testData.filter((item) => !item.disabled).map((data) => data.label);
     const actualColumns = await dataTable.getColumnHeadersText();
 
-    expect(actualColumns).toEqual(expectedColumns);
+    await expect(actualColumns).toEqual(expectedColumns);
```
Comment (adinapitul) [inline e2e/suites/extensions/ext-document-list.test.ts:87]: why do you add await here, and only in some places?
I'd say you either add it for every single expect from the project, or not at all.

**Review by adinapitul — CHANGES_REQUESTED** [review]

File: e2e/suites/extensions/ext-document-list.test.ts (hunk 1 above)
Comment (DenysVuika) [reply e2e/suites/extensions/ext-document-list.test.ts:87]: it's like saying "either test the whole app or don't test it at all" :) I would prefer the iteration approach and continuous improvements :) contributing just the pieces that making sad one of the code analysis tool I'm testing, this is not the ticket/task to change all

Comment (adinapitul) [reply e2e/suites/extensions/ext-document-list.test.ts:87]: true :)

maybe we can see if we can remove @types/jasminewd2 all together, and leave only jasmine types, that would solve the need for await in front of expect

File: projects/aca-testing-shared/src/utilities/repo-client/apis/repo-api.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ import { Logger } from '@alfresco/adf-testing';
 export abstract class RepoApi {
   alfrescoJsApi = new AlfrescoApi();
 
-  constructor(private username: string = browser.params.ADMIN_USERNAME, private password: string = browser.params.ADMIN_PASSWORD) {
+  protected constructor(private username: string = browser.params.ADMIN_USERNAME, private password: string = browser.params.ADMIN_PASSWORD) {
```
Comment (popovicsandras) [inline projects/aca-testing-shared/src/utilities/repo-client/apis/repo-api.ts:33]: Same question here.

File: src/app/components/page.component.ts [hunk 3]
```diff
@@ -70,7 +70,7 @@ export abstract class PageComponent implements OnInit, OnDestroy, OnChanges {
 
   protected subscriptions: Subscription[] = [];
 
-  constructor(protected store: Store<AppStore>, protected extensions: AppExtensionService, protected content: ContentManagementService) {}
+  protected constructor(protected store: Store<AppStore>, protected extensions: AppExtensionService, protected content: ContentManagementService) {}
```
Comment (popovicsandras) [inline src/app/components/page.component.ts:73]: Same here

**Review by popovicsandras — APPROVED** [review]

File: projects/aca-testing-shared/src/components/dialog/generic-dialog.ts [hunk 4]
```diff
@@ -27,7 +27,7 @@ import { ElementFinder, by, browser, Locator } from 'protractor';
 import { isPresentAndDisplayed, waitForPresence, waitForVisibility, waitForStaleness } from '../../utilities/utils';
 
 export abstract class GenericDialog {
-  constructor(private rootCssSelector?: string) {}
+  protected constructor(private rootCssSelector?: string) {}
```
[the remark this answers is not in the document]
Comment (DenysVuika) [reply projects/aca-testing-shared/src/components/dialog/generic-dialog.ts:30]: Just making the IntelliJ code analysis happy :)

File: projects/aca-testing-shared/src/utilities/repo-client/apis/repo-api.ts (hunk 2 above)
Comment (DenysVuika) [reply projects/aca-testing-shared/src/utilities/repo-client/apis/repo-api.ts:33]: it is a good practice making abstract constructors protected, following the tooling suggestions

**Pull request merged**